src/java/src/main/native/generator_params.cpp

```diff
@@ -0,0 +1,77 @@
+ * Licensed under the MIT License.
+ */
+#include <jni.h>
+#include "ort_genai_c.h"
+#include "utils.h"
+
+using namespace Helpers;
+
+extern "C" JNIEXPORT jlong JNICALL
+Java_ai_onnxruntime_genai_GeneratorParams_createGeneratorParams(JNIEnv* env, jobject thiz, jlong model_handle) {
+  const OgaModel* model = reinterpret_cast<const OgaModel*>(model_handle);
+  OgaGeneratorParams* generator_params = nullptr;
+  if (ThrowIfError(env, OgaCreateGeneratorParams(model, &generator_params))) {
+    return 0;
+  }
+
+  return reinterpret_cast<jlong>(generator_params);
+}
+
+extern "C" JNIEXPORT void JNICALL
+Java_ai_onnxruntime_genai_GeneratorParams_destroyGeneratorParams(JNIEnv* env, jobject thiz, jlong native_handle) {
+  OgaGeneratorParams* generator_params = reinterpret_cast<OgaGeneratorParams*>(native_handle);
+  OgaDestroyGeneratorParams(generator_params);
+}
+
+extern "C" JNIEXPORT void JNICALL
+Java_ai_onnxruntime_genai_GeneratorParams_setSearchOptionNumber(JNIEnv* env, jobject thiz, jlong native_handle,
+                                                                 jstring option_name, jdouble value) {
+  OgaGeneratorParams* generator_params = reinterpret_cast<OgaGeneratorParams*>(native_handle);
+  CString name{env, option_name};
+
+  ThrowIfError(env, OgaGeneratorParamsSetSearchNumber(generator_params, name, value));
+}
+
+extern "C" JNIEXPORT void JNICALL
+Java_ai_onnxruntime_genai_GeneratorParams_setSearchOptionBool(JNIEnv* env, jobject thiz, jlong native_handle,
+                                                               jstring option_name, jboolean value) {
+  OgaGeneratorParams* generator_params = reinterpret_cast<OgaGeneratorParams*>(native_handle);
+  CString name{env, option_name};
+
+  ThrowIfError(env, OgaGeneratorParamsSetSearchBool(generator_params, name, value));
+}
+
+extern "C" JNIEXPORT void JNICALL
+Java_ai_onnxruntime_genai_GeneratorParams_setInputSequences(JNIEnv* env, jobject thiz, jlong native_handle,
+                                                             jlong sequences_handle) {
+  OgaGeneratorParams* generator_params = reinterpret_cast<OgaGeneratorParams*>(native_handle);
+  const OgaSequences* sequences = reinterpret_cast<const OgaSequences*>(sequences_handle);
+
+  ThrowIfError(env, OgaGeneratorParamsSetInputSequences(generator_params, sequences));
+}
+
+extern "C" JNIEXPORT void JNICALL
+Java_ai_onnxruntime_genai_GeneratorParams_setInputIDs(JNIEnv* env, jobject thiz, jlong native_handle,
+                                                       jintArray token_ids, jint sequence_length, jint batch_size) {
+  OgaGeneratorParams* generator_params = reinterpret_cast<OgaGeneratorParams*>(native_handle);
+
+  auto num_tokens = env->GetArrayLength(token_ids);
+  jboolean is_copy = false;
+  jint* jtokens = env->GetIntArrayElements(token_ids, &is_copy);
+  if (is_copy) {
+    // we're dead as GenAI doesn't copy the inputs so the input data address will be invalid
```

Comment:
If it doesn't copy the memory then the Java side method should create a direct byte buffer, map it into an int buffer, write the array to it, store it in a field in the GeneratorParams and then pass the buffer down into JNI. The JNI code can then pull out the pointer from the buffer and hand it to the native code and Java will guarantee that the buffer lives as long as the Java GeneratorParams object does. This byte buffer dance is how ORT's tensors work, that's how we get zero copy through from Java into ORT. It'll look a bit like this - https://github.com/microsoft/onnxruntime/blob/main/java/src/main/java/ai/onnxruntime/OrtUtil.java#L507, and then on the JNI side - https://github.com/microsoft/onnxruntime/blob/main/java/src/main/native/ai_onnxruntime_OnnxTensor.c#L98.

What happens if this method is called multiple times is then something we need to decide the semantics for.